lib/elliptic/ec/signature.js

```diff
@@ -23,35 +23,76 @@ function Signature(options, enc) {
 }
 module.exports = Signature;
 
+function getLength(buf, p) {
+  var initial = buf[p.place++];
+  if (!(initial & 0x80)) {
+    return initial;
+  }
+  var octetLen = initial & 0xf;
+  var byteLength = octetLen;
+  var val = buf[p.place + --byteLength];
```

Comment:
done

On Wed, Oct 28, 2015 at 2:57 PM Fedor Indutny <notifications@github.com>
wrote:

> In lib/elliptic/ec/signature.js
> <https://github.com/indutny/elliptic/pull/58#discussion_r43301946>:
>
> > @@ -23,35 +23,76 @@ function Signature(options, enc) {
> >  }
> >  module.exports = Signature;
> >
> > +function getLength(buf, p) {
> > +  var initial = buf[p.place++];
> > +  if (!(initial & 0x80)) {
> > +    return initial;
> > +  }
> > +  var octetLen = initial & 0xf;
> > +  var byteLength = octetLen;
> > +  var val = buf[p.place + --byteLength];
>
> Any reason it can't be a loop like this?
>
> var val = 0;for (var i = 0, off = p.place; i < byteLength; i++, off++) {
>   val <<= 8;
>   val |= buf[off];
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/indutny/elliptic/pull/58/files#r43301946>.
>